shared/js/l10n.js

```diff
@@ -1161,19 +1178,27 @@
     }
 
     var idToFetch = this.isPseudo ? ctx.defaultLocale : this.id;
+    var source = navigator.mozL10n._config.localeSources[this.id] || 'app';
+    var gaiaVersion = navigator.mozL10n._config.gaiaVersion;
+
     for (var i = 0; i < ctx.resLinks.length; i++) {
       var resLink = decodeURI(ctx.resLinks[i]);
       var path = resLink.replace('{locale}', idToFetch);
       var type = path.substr(path.lastIndexOf('.') + 1);
 
+      var cb = null;
+      var ioType = 'text';
       switch (type) {
         case 'json':
-          io.loadJSON(path, onJSONLoaded, sync);
+          ioType = 'json';
```

Comment:
With this code, you're assigning to `ioType` twice in every loop pass if the type if `json`.  Can we just pass `type` here and if needed change `properties` to `text` in `bindingsIO.extra`?